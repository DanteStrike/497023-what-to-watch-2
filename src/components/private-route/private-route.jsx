import React from "react";
import PropTypes from "prop-types";

import {connect} from "react-redux";
import {Route, Redirect} from "react-router-dom";

import {userSelectors} from "../../reducers/user/user";


const PrivateRoute = (props) => {
  const {component: Component, isAuth} = props;

  const rest = Object.assign({}, props);
  delete rest.component;
  delete rest.isAuth;

  return (
    <Route
      {...rest}
      render={
        (routerProps) => (
          isAuth ?
            <Component {...routerProps}/>
            :
            <Redirect
              to={{
                pathname: `/login`,
                state: {referrer: rest.location.pathname ? rest.location.pathname : ``}
              }}
            />
        )
      }
    />
  );
};

PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired,
  isAuth: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  isAuth: userSelectors.getIsAuth(state)
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
