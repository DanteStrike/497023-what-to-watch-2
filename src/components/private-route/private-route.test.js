import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {PrivateRoute} from "./private-route";

it(`Component PrivateRouter should render correctly`, () => {
  const MockComponent = () => <div/>;

  const component = renderer.create(
      <BrowserRouter>
        <PrivateRoute component={MockComponent} isAuth={true} checkAuth={jest.fn()}/>
      </BrowserRouter>
  ).toJSON();

  expect(component).toMatchSnapshot();
});
