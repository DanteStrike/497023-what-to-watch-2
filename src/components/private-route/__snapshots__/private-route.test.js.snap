// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrivateRouter should render correctly 1`] = `<div />`;
